runtime/particle-execution-context.js

```diff
@@ -245,11 +245,11 @@ export class ParticleExecutionContext {
     });
 
     return [particle, async () => {
-      resolve();
-      let idx = this._pendingLoads.indexOf(p);
-      this._pendingLoads.splice(idx, 1);
       await particle.setHandles(handleMap);
       registerList.forEach(({proxy, particle, handle}) => proxy.register(particle, handle));
+      let idx = this._pendingLoads.indexOf(p);
+      this._pendingLoads.splice(idx, 1);
+      resolve();
```

Comment:
fyi this is the key piece: I think we need to await setHandles before calling resolve. It doesn't break any of the tests but I'm a bit concerned about this having unintended side effects...

Reply:
Hmm, looks like it might be breaking a test after all (the book-a-restaurant flow test).

Reply:
... or maybe not, seems to failing at ToT too. :-/

Reply:
Oh, hmm. Was that one of my changes that broke it?

I think we should try and figure out what's breaking in the morning before merging this, you OK with that?

Reply:
I think that was your change, but it was a long time ago .. I don't think we've ever tested "slow" firebase interactions in a unit test that exits quickly, and explicitly deletes the firebase App object (which was my change not long ago).

But yes, we'll have a deeper look before committing this.